### PR TITLE
[MODULAR] Loading diffirent types of ammo in multi-sprite mags and !POTENTIALY! in other ammo mags.

### PR DESCRIPTION
**Pull request by SSensum13**
## About The Pull Request

This PR adds possibility of loading different types of ammo in specialized ammo mags, by default it is turned on only for multi-sprite mags.

## How This Contributes To The Skyrat Roleplay Experience

Security no longer will have to abuse hacked autolathe or other mags to load lethal in their CMGs.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
218198869-32c75e86-8868-4916-9e8d-7fa6832de415

</details>

## Changelog

:cl:
qol: Security no longer have to abuse hacked autolathe to load other types of ammo in their CMGs and other multi-sprite weapons.
code: Added support of "multitype" variable for ammo_workbench. 
/:cl:

### CODE REVIEW

**Comment by SSensum13** [comment]
Ugh... Whats wrong? How do i fix that test result?

**Comment by masline147** [comment]
Well, finally, the machine will no longer be required for each shift. It will also be easier for the sergeant, because he will no longer have to remove the brig from the casings.

**Comment by ROKBAS** [comment]
Very interested in this feature. This make sense, because of ammo size, not types.

**Comment by Paxilmaniac** [comment]
> Ugh... Whats wrong? How do i fix that test result?

The screenshot tests failing is not your doing, don't worry about it.

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 1]
```diff
@@ -35,6 +35,11 @@
 	var/list/bullet_cost
 	///cost of the materials in the magazine/box itself
 	var/list/base_cost
+	
+	//SKYRAT EDIT ADDITION BEGIN
+	///allows loading other types of ammo via ammo_workbench
+	var/multitype = FALSE
+	//SKYRAT EDIT ADDITION ENDS
```
Comment (tf-4) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm]: can do this in a modular file

**Comment by tf-4** [comment]
looks good tho

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm (hunk 1 above)
Comment (SSensum13) [reply code/modules/projectiles/boxes_magazines/_box_magazine.dm]: i will do that tomorrow

**Comment by SSensum13** [comment]
@tf-4, i made it fully modular.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**